openshift/templates/email-verification-service/email-verification-service-deploy.overrides.sh

```diff
@@ -1,67 +1,31 @@
+#! /bin/bash
+_includeFile=$(type -p overrides.inc)
+# Import ocFunctions.inc for getSecret
+_ocFunctions=$(type -p ocFunctions.inc)
+if [ ! -z ${_includeFile} ]; then
+  . ${_ocFunctions}
+  . ${_includeFile}
+else
+  _red='\033[0;31m'; _yellow='\033[1;33m'; _nc='\033[0m'; echo -e \\n"${_red}overrides.inc could not be found on the path.${_nc}\n${_yellow}Please ensure the openshift-developer-tools are installed on and registered on your path.${_nc}\n${_yellow}https://github.com/BCDevOps/openshift-developer-tools${_nc}"; exit 1;
+fi
 # ===================================================================================
 # Special Deployment Parameters needed for the email-verification-service instance.
 # -----------------------------------------------------------------------------------
 # ===================================================================================
-printStatusMsg(){
-  (
-    _msg=${1}
-    _yellow='\033[1;33m'
-    _nc='\033[0m' # No Color
-    printf "\n${_yellow}${_msg}\n${_nc}" >&2
-  )
-}
 
-readParameter(){
-  (
-    _msg=${1}
-    _paramName=${2}
-    _defaultValue=${3}
-    _encode=${4}
 
-    _yellow='\033[1;33m'
-    _nc='\033[0m' # No Color
-    _message=$(echo -e "\n${_yellow}${_msg}\n${_nc}")
 
-    read -r -p $"${_message}" ${_paramName}
-
-    writeParameter "${_paramName}" "${_defaultValue}" "${_encode}"
-  )
-}
-
-writeParameter(){
-  (
-    _paramName=${1}
-    _defaultValue=${2}
-    _encode=${3}
-
-    if [ -z "${_encode}" ]; then
-      echo "${_paramName}=${!_paramName:-${_defaultValue}}" >> ${_overrideParamFile}
-    else
-      # The key/value pair must be contained on a single line
-      _encodedValue=$(echo -n "${!_paramName:-${_defaultValue}}"|base64 -w 0)
-      echo "${_paramName}=${_encodedValue}" >> ${_overrideParamFile}
-    fi
-  )
-}
-
-initialize(){
-  # Define the name of the override param file.
-  _scriptName=$(basename ${0%.*})
-  export _overrideParamFile=${_scriptName}.param
-
-  printStatusMsg "Initializing ${_scriptName} ..."
-
-  # Remove any previous version of the file ...
-  if [ -f ${_overrideParamFile} ]; then
-    printStatusMsg "Removing previous copy of ${_overrideParamFile} ..."
-    rm -f ${_overrideParamFile}
-  fi
-}
 
 initialize
 
-# Get the webhook URL
-readParameter "EMAIL_HOST - Please provide the host name of the email server:" EMAIL_HOST "smtp.host.io" 
+if createOperation; then
+  # Get the webhook URL
+  readParameter "SMTP_EMAIL_HOST - Please provide the host name of the email server:" SMTP_EMAIL_HOST "smtp.host.io" "false"
+  else
+  # Secrets are removed from the configurations during update operations ...
+  printStatusMsg "Update operation detected ...\nSkipping the prompts for SMTP_EMAIL_HOST secret... \n"
```

Comment:
It's good form to indicate the value for the variable is been read back from a secret, for example; https://github.com/bcgov/von-bc-registries-agent-configurations/blob/feature/ocp4/openshift/templates/event-processor/event-processor-deploy.overrides.sh#L20

Reply:
updated status msg